docs/content/en/integration/openid-connect/roundcube/index.md

```diff
@@ -0,0 +1,231 @@
+applies. However, if you want to use the *same* reverse-proxy instance for
+your protected application, its corresponding vhost might need the following
+adaptations:
+
+```nginx
+# Backend proxied app on the same host
+server {
+    # See N.B. below
+    listen 9092;
+    server_name roundcube;
+
+    ...
+
+    location ~ [^/]\.php(?:/|$) {
+        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
+        if (!-f $document_root$fastcgi_script_name) {
+            return 404;
+        }
+
+        include fastcgi_params;
+
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+        fastcgi_param PATH_INFO       $fastcgi_path_info;
+
+        # See N.B. below
+        fastcgi_param  SERVER_PORT '';
+
+        fastcgi_pass $your_php_upstream;
+    }
+}
+```
+
+**N.B.** Roundcube's redirect URI is _not_ configurable, but is
+dynamically built with bits coming from the FCGI environment:
+`<scheme>://<fqdn>[:<port>]/...`. To force HTTPS, Roundcube's conf flag
+`use_https` must be set. However, the redirection breaks when the backend
+application is listening on a specific port, because the resulting redirect
+URI would be something like `https://<fqdn>:<port>/...`. Thus, to obtain the
+correct redirect URI `'https://<fqdn>/...'`, the fastcgi_param SERVER_PORT
+must be unset.
+
+
+### Dovecot
+
+[Dovecot OAuth2] configuration goes into two files.
+
+#### Common configuration
+
+Normally in file `/etc/dovecot/dovecot.conf` or one of its ancillary files in
+`/etc/dovecot/conf.d/`:
+
+```nginx
+auth_mechanisms = $auth_mechanisms oauthbearer xoauth2
+
+passdb {
+  args = /etc/dovecot/dovecot-oauth2.conf.ext
+  driver = oauth2
+  mechanisms = xoauth2 oauthbearer
+}
+
+# Optional for Postfix SASL on smtpd/submission
+service auth {
+  unix_listener /var/spool/postfix/private/auth {
+    group = postfix
+    mode = 0666
+    user = postfix
+  }
+}
+```
+
+#### Backend configuration
+
+As defined above, in file,  `/etc/dovecot/dovecot-oauth2.conf.ext`:
+
+```nginx
+introspection_mode = post
+introspection_url = https://roundcube:insecure_secret@auth.example.com/api/oidc/introspection
+username_attribute = username
+```
+
+**N.B.** The client ID and secret must figure as credentials in the `introspection_url`.
+
+### Postfix
+
+Even though no authentication would be required when your Postfix instance is
+on the same host, Roundcube OAuth2 [enforces 'XOAUTH2' auth type plus
+credentials](https://github.com/roundcube/roundcubemail/issues/9183) and gives
+up the SMTP + SSL/TLS handshaking as no auth options would be offered from
+Postfix. Thus, Postfix must be configured with (Dovecot-type)
+[SASL](https://www.postfix.org/SASL_README.html) on port 25 (smtpd) or 587
+(submission), with the following minimum set of options:
+
+```nginx
+smtpd_sasl_auth_enable = yes
+smtpd_sasl_path = private/auth
+smtpd_sasl_security_options = noanonymous, noplaintext
+smtpd_sasl_tls_security_options = noanonymous
+smtpd_sasl_type = dovecot
+```
+
```

Comment:
I love all the extra details here. However this makes me think we should move all Authelia configuration sections above the application itself. One it makes sense Authelia must be configured first and all the advanced things like this can then be in one area which is implementation specific. Could you reflect this idea in your change? i.e. have `### Authelia` directly under `## Configuration` and make `### Application` etc. be after `### Authelia`?

Reply:
Sure!